### DISCUSSION TITLE
Pouvoir simuler des CDD inférieurs à un mois

### DISCUSSION THREAD

**Discussion by johangirod** [betagouv/mon-entreprise#2604] [Idées]
La simulation ne permet pas de rentrer une durée inférieur à un mois pour les CDD. Or ceux-ci peuvent être fréquemment utilisé, notamment dans le cadre d'un accroissement temporaire d'activité.

- [ ] #1696
- [ ] Ajouter un cas dans le contrôle du SMIC qui vérifie si le contrat est un CDD < 1 mois
- [ ] Vérifier quelles règles s'appliquent pour la mutuelle (prorata ? exception si < à un certain temps ?)
- [ ] (facultatif) Pouvoir changer l'échelle de temps du champ (jour, mois, année)

**Comment by bobylito** [comment]
J'ai pu [créer un contrat de 0.3 mois](https://mon-entreprise.fr/simulateurs/salaire-brut-net?contrat+salarié='CDD'&contrat+salarié+.+ATMP+.+taux+connu=non&contrat+salarié+.+ATMP+.+taux+réduit=oui&contrat+salarié+.+CDD+.+contrat+jeune+vacances=non&contrat+salarié+.+CDD+.+durée+contrat=0.3mois&contrat+salarié+.+CDD+.+motif='classique+.+mission'&contrat+salarié+.+CDD+.+événement=non&contrat+salarié+.+aides+employeur+.+aide+à+l'embauche+des+travailleurs+handicapés+.+situation+de+handicap=non&contrat+salarié+.+ancienneté+.+date+d'embauche='01%2F01%2F2021'&contrat+salarié+.+complémentaire+santé+.+forfait=40€%2Fmois&contrat+salarié+.+complémentaire+santé+.+part+employeur=100%25&contrat+salarié+.+convention+collective='droit+commun'&contrat+salarié+.+déduction+forfaitaire+spécifique=non&contrat+salarié+.+frais+professionnels+.+abonnement+transports+publics+.+montant=75€%2Fmois&contrat+salarié+.+frais+professionnels+.+titres-restaurant=non&contrat+salarié+.+régime+des+impatriés=non&contrat+salarié+.+rémunération+.+primes+.+fin+d'année+.+treizième+mois=non&contrat+salarié+.+statut+JEI=non&contrat+salarié+.+statut+cadre=non&contrat+salarié+.+temps+de+travail+.+temps+partiel=non&cout-embauche=100€%2Fmois&entreprise+.+association+non+lucrative=non&entreprise+.+effectif+.+seuil='moins+de+5'&entreprise+.+exonérée+de+TVA=non&impôt+.+méthode+de+calcul='taux+neutre'&situation+personnelle+.+domiciliation+fiscale+à+l'étranger=non&utm_source=sharing)

La partie qui indique le temps de travail continue d'indiquer un nombre d'heures par mois. Cela ne m'a pas l'air faux en soit si on a 0.3mois de travail... Peut-être que cela devrait indiquer le nombre d'heures total et pas par mois si on utilise un chiffre inférieur à 0?

<img width="869" alt="Screenshot 2021-09-08 at 12 27 13" src="https://user-images.githubusercontent.com/393765/132493705-05161f5d-b9ba-4869-8d2d-51e75cbbf398.png">

Le total est le même que si on avait 1 mois complet par contre...

**Comment by lajarre** [comment]
> Le total est le même que si on avait 1 mois complet par contre...

Oui c'est un peu confusant. On pourrait juste ne pas afficher cette ligne dans le cas `<1`, je pense. Mais il faudrait idéalement voir à quoi ressemble une fiche de paie pour un contrat court.

**Comment by lajarre** [comment]
> J'ai pu créer un contrat de 0.3 mois

C'est vrai, mais ça va pas être évident pour l'utilisateur de raisonner ainsi. Un quick fix serait de proposer qqch du genre "1 semaine", "2 semaines" et "3 semaines" en suggestion (aujour'dhui il y a 18 mois, 1 an, etc.), et que ce truc injecte le bon nombre décimal qu'il faut.

Un truc mieux serait de proposer un switch pour passer en jours, lorsqu'on clique sur "mois" par exemple. (on peut proposer semaines aussi)  
L'idée derrière est d'émettre la bonne action redux en remplaçant `mois` par `jour`, et ça devrait marcher tout seul ensuite

<img width="254" alt="image" src="https://user-images.githubusercontent.com/1884912/132539555-78bd7bf7-68b7-493b-9b38-3b3b0cfbceb1.png">

**Comment by mquandalle** [comment]
> Mais il faudrait idéalement voir à quoi ressemble une fiche de paie pour un contrat court.

Le nombre d'heures travaillées ainsi que le taux de rémunération horaire sont notés. Rien de particulier sinon https://www.cogilog.com/bulletin-cddinf/

**Comment by lajarre** [comment]
oui donc mettre temps de travail en heures total et non pas en heures/mois.

**Comment by glopezrios** [comment]
Le besoin a de nouveau été exprimé par un usager, pour un contrat d'usage d'une seule journée.

Le simulateur prend bien en compte le contrat d'usage mais il est compliqué de transformer un jour en fraction de mois.
Et sur la partie "fiche de paie", le temps de travail reste toujours un total d'heures par mois.

**Comment by glopezrios** [comment]
La question de pouvoir entrer une durée inférieure à un mois rejoint [la discussion #2706](https://github.com/betagouv/mon-entreprise/discussions/2706) (coût à la journée), qui elle-même a un lien avec [l'issue #475](https://github.com/betagouv/mon-entreprise/issues/475) (taux horaire).

**Comment by glopezrios** [comment]
A prolonger dans #3015
